src/TwentyTwenty.Storage.Local/LocalStorageProvider.cs

```diff
@@ -244,21 +264,100 @@ public Task<IList<BlobDescriptor>> ListBlobsAsync(string containerName)
                 {
                     source.Dispose();
                 }
+                if (properties != default)
+                {
+                    await UpdateBlobPropertiesAsync(containerName, blobName, properties);
+                }
             }
             catch (Exception ex)
             {
                 throw ex.ToStorageException();
             }
         }
-
+        
         public void UpdateBlobProperties(string containerName, string blobName, BlobProperties properties)
         {
-            return;
+            UpdateBlobPropertiesAsync(containerName, blobName, properties).ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Lets create a non-async version of `UpdateBlobProperties`, so we don't have to use `GetResult()`. `StreamWriter` has `Write` and `Flush` methods, to replace the async ones.